api/src/main/java/jakarta/servlet/ServletOutputStream.java

```diff
@@ -51,6 +53,37 @@ public abstract class ServletOutputStream extends OutputStream {
     protected ServletOutputStream() {
     }
 
+    /**
+     * Writes from the given buffer to the output stream. The method blocks until all the remaining data in the buffer has
+     * been written.
+     * <p>
+     * If this method is called when the buffer has no data remaining, the method returns immediately.
+     * <p>
+     * When the method returns, and if data has been written, the buffer's limit will be unchanged from the value when
+     * passed to this method and the position will be the same as the limit.
+     * <p>
```

Comment:
This may not be true for asynchronous writes.  It is true for asynchronous write once this method has returned AND `isReady()` has returned true.

Reply:
More over, callers must not modify the position, limit nor the underlying buffer content until `isReady()` has returned true.

Reply:
Good points. I based the text on the existing [In|Out]putStream text which doesn't address non-blocking. I'll update the descriptions.